WordPress/two-factor
# Delete user meta on plugin uninstall

## What?

Fixes #630.

## Why?

Delete all used user meta keys and options during plugin uninstall as expected.

## How?

1. Register an [uninstall hook](https://developer.wordpress.org/reference/functions/register_uninstall_hook/) for the plugin.

2. During uninstall, collect all meta keys and options used by two-factor providers through a known class helper. Delete them by looping through each user (a performance concern on larger sites).

Consider using a direct database query to purge the user meta by keys to avoid looping through every user on larger sites.

## Testing Instructions

1. Setup the plugin with a few users and two-factor methods enabled for several users.
2. Uninstall the plugin (in addition to deactivating).
3. Activate the plugin again and confirm that no settings for providers have been preserved.

## Screenshots or screencast


## Changelog Entry


Added: register an uninstall hook for the plugin to delete all user meta data when the plugin is completely uninstalled.
